cogs/strawpoll.py

```diff
@@ -0,0 +1,458 @@
+# Strawpoll functionality by Savestate for Red-DiscordBot
+
+import urllib3
+import certifi
+import json
+import traceback
+import discord
+import asyncio
+import os
+from collections import defaultdict
+from .utils import checks
+from .utils.dataIO import dataIO
+from concurrent.futures import CancelledError
+from html import unescape
+from time import sleep, time, strftime
+from discord.ext import commands
+
+def _get_poll(poll_id):
+    # https://strawpoll.me/api/v2/polls/{poll_id}
+    https = urllib3.PoolManager(
+        cert_reqs='CERT_REQUIRED',
+        ca_certs=certifi.where())
+    try:
+        request = https.request( 'GET', 
+            'https://strawpoll.me/api/v2/polls/' + str(poll_id),
+            retries=3, timeout=3.0)
+        json_response = json.loads(request.data.decode('utf-8'))
+    except urllib3.exceptions.HTTPError:
+        return None
+    except json.decoder.JSONDecodeError:
+        return None
+    return json_response
+    
+def _post_poll(title, options, multi):
+    # build json request
+    data = {
+        'title': title,
+        'options': options,
+        'multi': multi,
+        'dupcheck': 'normal',
+        'captcha': False }
+    encoded_data = json.dumps(data).encode('utf-8')
+    # urllib3 pool manager in https mode
+    https = urllib3.PoolManager(
+        cert_reqs='CERT_REQUIRED',
+        ca_certs=certifi.where())
+    try:
+        request = https.request(
+            'POST', 'https://strawpoll.me/api/v2/polls',
+            body=encoded_data, headers={
+                'Content-Type': 'application/json',
+                'Accept-Charset': 'utf-8'
+            },
+            retries=3, timeout=3.0)
+        json_response = json.loads(request.data.decode('utf-8'))
+    except urllib3.exceptions.HTTPError:
+        return None
+    return json_response
+
+class _Poll:
+    """Internal poll class for Strawpoll cog"""
+    
+    def __init__(self, bot, settings, message, 
+            author, poll_id, poll_length, sleep_time):
+        self.bot = bot
+        self.settings = settings
+        self.message = message
+        self.author = author
+        self.poll_id = poll_id
+        self.poll_title = ""
+        self.poll_length = poll_length
+        self.sleep_time = sleep_time
+        self.start_time = time()
+
+    # https://www.w3resource.com/python-exercises/python-basic-exercise-65.php
+    def _time_left(self, seconds):
+        days = seconds // (24 * 3600)
+        seconds = seconds % (24 * 3600)
+        hours = seconds // 3600
+        seconds %= 3600
```

Comment:
Didn't know you could do mod-equals in python. Took me a minute to realize what this meant lol